app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,64 @@
+class OrdersController < ApplicationController
+  def index
+    @orders = Order.for_user(current_user)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def no_cart
+    flash[:alert] = 'you have not put anything in the cart'
+    redirect_to root_path
+  end
+
+  def create
+    if current_user.present?
+      order = find_order
+      if order.blank?
+        params[:alert] = "This Order is not belongs to you"
+        redirect_to :back
+        return
+      end
+
+      if !params[:order_detail][:quantity].to_s.match(/\A[+-]?\d+?(\.\d+)?\Z/) || params[:order_detail][:quantity].to_i < 0
```

Comment:
do you have any better way to validate params to be number and must be greater than 0?